### ISSUE TITLE
Missing lock for job failed / Missing process handler for job type

### ISSUE THREAD

**Issue by orlandosantos** [OptimalBits/bull#700]
Hi,

I'm trying to use processJob function but I've got the error message: "Error: Missing lock for job 56 failed" and when I look at the reason of failure of the job i've got the message: Error: Missing lock for job 56 failed.

Could you please send me a example how I use the processJob function?

Note: I'm trying to use this function after I got job with getJob. I need to complete a specific job and don't know a better soluction to do this.

Thanks

**Comment by orlandosantos** [comment]
Sorry, the message received on reason of failure is "Missing process handler for job type __default__"

**Comment by manast** [comment]
there are examples in the readme. If you still cannot figure it out please paste your code here and I will help you out.

**Comment by orlandosantos** [comment]
please find the code below;

	var Queue    = require('bull');
	var config   = require("./config");

	var ariQueue = new Queue('ari', {redis: {port: config.queue.port, host: config.queue.host}});
	var jobID = 108;

	ariQueue.getJob(jobID).then(function(job){
		ariQueue.processJob(job).then(function(result){
			console.log(result);
		})
	});

**The result is;**

Unhandled rejection Error: Missing lock for job 108 failed
    at Object.finishedErrors (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bull\lib\scripts.js:148:23)
    at P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bull\lib\job.js:213:31
    at bound (domain.js:280:14)
    at runBound (domain.js:293:12)
    at tryCatcher (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\promise.js:512:31)
    at Promise._settlePromise (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\promise.js:569:18)
    at Promise._settlePromise0 (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\promise.js:614:10)
    at Promise._settlePromises (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\promise.js:693:18)
    at Async._drainQueue (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\async.js:133:16)
    at Async._drainQueues (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\async.js:143:10)
    at Immediate.Async.drainQueues (P:\Documents\Pactual\Projetos\VICCI\GIT\vicci\telephone\core\node_modules\bluebird\js\release\async.js:17:14)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)

**The reason of failure on job 108 is:** Missing process handler for job type __default__

Thanks for your help!

**Comment by manast** [comment]
you are not supposed to call ```processJob``` directly, that is a private method. You just need to define your process function calling ```process```: https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#queueprocess

**Comment by orlandosantos** [comment]
Thanks for your response!

It is possible use process function to process a specific job, using id?

**Comment by manast** [comment]
nop

**Comment by yangju** [comment]
I got the Missing process Handler for job type issue too.
It works locally (sandboxed)
queue.process(2, path.resolve('job/JobProcessor.js'));
or 
queue.process(2, path.join(__dirname, 'JobProcessor.js'));
However, when I deployed the application to AWS (fargate hosted docker container), it throws  Missing process Handler for any job it handles. I can see the jobs are marked as failed in redis.
If I make it in process such as queue.process(function(job){.....}, it works in aws.
Maybe the application could not find the 'JobProcessor.js' in a serverless fargate deployed container?
